cli/util/upload_bin.go

```diff
@@ -32,8 +32,12 @@ import (
 )
 
 const (
-	// Number of objects to search in a batch for finding max block in container.
-	searchBatchSize = 10000
+	// Number of objects to upload in a batch.
+	uploadBatchSize = 10000
+	// Number of objects to search in a batch.
+	// For searching the oldest missing block it should be equal to uploadBatchSize.
+	// But due to incomplete search results #3645 the object.MatchStringEqual is used.
```

Comment:
> For searching the oldest missing block it should be equal to uploadBatchSize.

It's not exactly what I'd like to see here. Write about the reason instead of describing the code behaviour. In two months I'll forget about why do we require it to be equal to uploadBatchSize. Write something about the fact that this code relies on the fact that uploader is able to create only *a single (latest)* incomplete batch of 10K objects, whereas the rest of batches are expected to be completed.